pkg/adsc/delta.go

```diff
@@ -426,7 +548,9 @@ func (c *Client) handleDeltaResponse(d *discovery.DeltaDiscoveryResponse) error
 		return nil
 	}
 	for _, r := range d.Resources {
-		if d.TypeUrl != r.Resource.TypeUrl {
+		_, isMcp := convertTypeURLToMCPGVK(d.TypeUrl)
```

Comment:
why is the `d.TypeUrl != r.Resource.TypeUrl` for mcp?

Reply:
hm looks like we send `type.googleapis.com/istio.mcp.v1alpha1.Resource`.

Can we make this check just compare to that string? `convertTypeURLToMCPGVK` is not super cheap to run on every object

Reply:
This is a reference to adsc's handling of the `TypeUrl` in the response.
In addition, the `APIGenerator` has a note stating that the `TypeUrl` is of the form `g/v/k` for the mcp resource xds response.

https://github.com/istio/istio/blob/1f96743ef2c31e06b3915947182781a28e05614a/pilot/pkg/networking/apigen/apigen.go#L65-L71

Limited to, multiple different gvk's, we don't seem to have a way to directly compare strings yet.